DominoSwissWeatherstation/module.php

```diff
@@ -41,11 +40,37 @@ public function ReceiveData($JSONString) {
 		if ($data->Values->ID == $this->ReadPropertyInteger("ID")) {
 			switch ($data->Values->Command) {
 				case 32:
-					SetValue($this->GetIDForIdent("LightValue"), $this->GetLightValue(intval($data->Values->Value / 8), ($data->Values->Value % 8)));
+					$newStep = intval($data->Values->Value / 8);
+					$newValue = $this->GetSimpleLightValue($newStep);
+
+					$lightId = $this->GetIDForIdent("LightValue");
+					$oldValue = GetValue($lightId);
+					$oldStep = $this->GetStepFromLightValue($oldValue);
+					if ($oldStep == -1) {
+						IPS_LogMessage("Invalid light value", "Old: " . $oldValue);
+						break;
+					}
+					
+					$delta = $this->ReadPropertyInteger("MaxLightStepDelta");
+					// We check that not more than $delta steps are changed at once. This is a simple way to avoid wrong values.
+					if (abs($newStep - $oldStep) > $delta) {
+						IPS_LogMessage("Unreasonable light step change", "Old: " . $oldValue . " New: " . $newValue . " Delta: " . $delta);
+					} else {
+						SetValue($lightId, $newValue);
+					}
 					break;
 					
 				case 33:
-					SetValue($this->GetIDForIdent("WindValue"), $this->GetWindValue(intval($data->Values->Value / 8), ($data->Values->Value % 8)));
+					$newValue = $this->GetSimpleWindValue(intval($data->Values->Value / 8));
+					$windId = $this->GetIDForIdent("WindValue");
+					$oldValue = GetValue($windId);
+					$delta = $this->ReadPropertyInteger("MaxWindValueDelta");
+					// We check that not more than $delta value is changed at once. This is a simple way to avoid wrong values.
```

Comment:
Missing the check for value of -1

Reply:
Either add the check, or change the comparisons in `GetStepFromWindValue()`/`GetStepFromLightValue()` to something like:
```
	private function GetStepFromWindValue($value) {
		foreach (self::VALID_WIND_VALUES as $key => $val) {
			if ($val >= $value) {
				return $key;
			}
		}
		return 0;
	}
```
So it never returns -1

Reply:
No, here its correct. For the wind we dont compare the jump in steps but the actual values.  But I have added a new check if the step value for the new wind value is valid.
And since `GetStepFromWindValue` isn't actually used I'll delete it again to reduce confusion